src/py/flwr/client/app.py

```diff
@@ -243,6 +268,10 @@ class `flwr.client.Client` (default: None)
         If set to None, there is no limit to the total time.
     flwr_path: Optional[Path] (default: None)
         The fully resolved path containing installed Flower Apps.
+    isolate: Optional[bool] (default: False)
```

Comment:
I suggest changing this to `run_type` and having it as an ENUM.

- internal (default): same process
- subprocess: basically what "isolate" does
- docker: in future PR's for now this would not be included here

Down the line we could refactor the whole thing and change it into a plugin system to support custom run types like e.g. K8, Lambda Functions, etc.